cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -202,6 +209,9 @@ protected void shutDown() throws Exception {
     runRecordCorrectorService.stopAndWait();
     pluginService.stopAndWait();
     provisioningService.stopAndWait();
+    if (runCountUpgradeService != null && runCountUpgradeService.isRunning()) {
```

Comment:
is there a reason we need to check it's running?

Reply:
Since we will stop the service when the upgrade is done: https://github.com/caskdata/cdap/blob/e78ed868fda433b7c8c79a25cd405c35de8c35a7/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/RunCountUpgradeService.java#L64